### DISCUSSION TITLE
Очистка кэша картинок будет?

### DISCUSSION THREAD

**Discussion by ghost** [maxsite/cms#446] [General]
Нет нормального способа очистить кэш картинок одним махом.
В openCart это реализовано шикарно. Можно отдельно чистить кэш картинок и кэш сайта. В Maxsite нужно ухитрятся и все делать либо ручками либо ручками особенно если статей за 100+ это тратит много времени.

Хотел спросить есть ли в планах реализация очистки кэша картинок?

**Comment by maxsite** [comment]
Что такое «кэш картинок»?

**Reply by ghost** [reply]
Доброго времени суток Максим. В Maxsite -  миниатюры. К сожалению их нельзя очистить нажатием одной кнопки.

**Reply by maxsite** [reply]
Это которые в mini-каталогах `uploads\_pages\77\mini\` ?

**Reply by ghost** [reply]
Да верно.

**Reply by maxsite** [reply]
Никогда об этом не задумывался... При очистке кэша происходит удаление пустых каталогов в _pages. В принципе можно удалять и файлы в mini, но после этого возникнет нагрузка на сервер, поскольку файлы нужно будет заново создавать.

Технически, я думаю, сделать можно. Наверное только будет отдельная опция или даже кнопка.

**Reply by ghost** [reply]
Да я с вами согласен нюансов много. Сейчас попытаюсь максимально практично изложить мысли)

**Reply by maxsite** [reply]
На самом деле, там всё просто. Удаляются все файлы в _pages, где в имени есть `/mini/`. Там в общем-то только изображения, так что ничего больше не затронет, проблем не должно возникнуть. Хотя, если шаблон не меняется, то  особой выгоды такая очистка не принесёт. Разве что старые записи, у которых нет посещений, миниатюры не будут создаваться.

**Comment by ghost** [comment]
У нас есть миниатюры картинок для каждого поста и не только(если юзать файловый-менеджер). Естественно по нажатию одной кнопки их не удалить. С постами это работает просто: можно изменить размер выводимой картинки на 1 пиксель и у нас треш с кучей файлов в папке **/mini/...** . В opencart картинки лежат в одной папке(исходники), а миниатюры в другой и создают подпапки. Исходник никогда не теряется и кэш картинок чистится нажатием одной кнопки, она чистит только картинки

**Reply by maxsite** [reply]
Это понятно. Удалить файлы в `mini` не проблема. Проблема возникнет, если загружать изображение с созданием миниатюры. То есть будет так:

```
1.jpg
mini/
    1.jpg
```

То есть миниатюра формируется именно в момент загрузки. Если её удалить, то она окажется недоступной по mini-url-адресу.

А вот те миниатюры, которые формируются автоматом (через thumb_generate), проблем не будет — они автоматом создадутся заново.

Так что есть риск, что после такой очистке, масса нужных файлов исчезнут.

**Reply by maxsite** [reply]
То есть я к тому, что нужно подумать, как отличать автоматически созданные файлы миниатюр, от созданных при загрузке. При обходе нужно будет проверять имя файла в основном _pages-страницы.

**Reply by ghost** [reply]
Я понимаю, что тут есть нюансы и вам определенно виднее какой путь выбрать. Я вот полагаю тут можно разделять и властвовать) Картинки миниатюры для админки в одной папке и подпапках, те что на фронте и через  thumb_generate в другом месте и чистим через кэш картинок и.т.д... В любом случае, благодарю за дискуссию и буду рад маленьким переменам в системе которые облегчят жизнь мне или кому-то еще)

**Reply by maxsite** [reply]
Сейчас уже разделять каталоги не имеет смысла — это нарушит совместимость и точно приведёт к проблемам. Да и разницы между thumb_generate  и обычным mini — нет: по сути это одно и тоже.

В любом случае подумаю, как это можно сделать. У меня самого куча таких файлов, правда они мелкие, особой роли не играют, но согласен, что порядок нужен. :)
